### PR TITLE
feat: new Prometheus metrics build_info

### PR DESCRIPTION
**Pull request by nlamirault**
Fixes #TODO

### CODE REVIEW

**Comment by Joibel** [comment]
I had proposed adding this in #12589. I am hoping to get that in for the next release, so this may be redundant work.

If you'd like to carry on with it please fill out the PR description, and add this to the documentation.

I have also included all of the values from GetVersion(), and not derived through other means. For compatibility with this I'd like it to match something like this.

```
	labelBuildVersion      string = `version`
	labelBuildPlatform     string = `platform`
	labelBuildGoVer        string = `goversion`
	labelBuildDate         string = `build`
	labelBuildCompiler     string = `compiler`
	labelBuildGitCommit    string = `commit`
	labelBuildGitTreeState string = `treestate`
	labelBuildGitTag       string = `tag`
```
...
```
	{name: labelBuildVersion, value: version.Version},
	{name: labelBuildPlatform, value: version.Platform},
	{name: labelBuildGoVer, value: version.GoVersion},
	{name: labelBuildDate, value: version.BuildDate},
	{name: labelBuildCompiler, value: version.Compiler},
	{name: labelBuildGitCommit, value: version.GitCommit},
	{name: labelBuildGitTreeState, value: version.GitTreeState},
	{name: labelBuildGitTag, value: version.GitTag},
```

File: workflow/metrics/server.go [hunk 1]
```diff
@@ -150,3 +165,9 @@ func (m *Metrics) garbageCollector(ctx context.Context) {
 		}
 	}
 }
+
+// recordBuildInfo publishes information about Argo-Rollouts version and runtime info through an info metric (gauge).
```
Comment (agilgur5) [inline workflow/metrics/server.go:169]: this isn't Argo Rollouts
```suggestion
// recordBuildInfo publishes metadata about Argo's version and runtime through a metric (gauge).
```

File: workflow/metrics/server.go [hunk 2]
```diff
@@ -14,9 +15,20 @@ import (
...
 
+	"github.com/argoproj/argo-workflows/v3"
 	tlsutils "github.com/argoproj/argo-workflows/v3/util/tls"
 )
 
+var (
+	buildInfo = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "build_info",
+			Help: "A metric with a constant '1' value labeled by version from which Argo-Workflows was built.",
```
Comment (agilgur5) [inline workflow/metrics/server.go:26]: and is not referred to as "Argo-Workflows" within the codebase or docs (generally speaking). 

this should also be consistent with the comment

```suggestion
			Help: "A metric with a constant '1' value labeled with metadata about Argo's version and runtime.",
```

**Review by agilgur5 — COMMENTED** [review]
some minor modifications below to the current code, otherwise Alan can do follow-on reviews

**Comment by agilgur5** [comment]
Superseded by #13269

**Pull request closed without merging**